test/ruby/test_zjit.rb

```diff
@@ -1675,6 +1675,20 @@ def self.test = @@x
     }
   end
 
+  def test_getclassvariable_raises
+    assert_compiles '"uninitialized class variable @@x in Foo"', %q{
+      class Foo
+        def self.test = @@x
+      end
+
+      begin
+        Foo.test
+      rescue NameError => e
+        e.message
+      end
+    }
```

Comment:
this should maybe do `compile_threshold=1` or whatever to make sure it actually gets compiled on the first go

Reply:
Oh ok yeah that works!